packages/react-server/core/ReactServerAgent/Request.js

```diff
@@ -233,7 +228,35 @@ function buildSuperagentRequest() {
 		}
 	}
 
-	req.send(postParams);
+	let hasQueryParams = (this._queryParams.length > 0),
+		hasPostParams = (Object.keys(this._postParams).length > 0);
+
+	if (hasPostParams) {
+		// superagent has some weird, implicit file upload support
+		// that only works if you don't set `type`.
+		if (this._type && this._type !== 'form-data') {
+			req.type(this._type);
+		}
+
+		req.send(postParams);
+	}
```

Comment:
I think there might be an edge case here. It seems like one should be able to POST an empty JSON object to an endpoint, and I think we can't anymore. (Though to be fair, I think that may have been the default before, which probably also doesn't make sense?)

I suspect that there have always been lots of edge cases here that we just haven't run into yet. (I just realized that superagent accepts strings _and_ objects for `send`, and we only accept objects.